Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Specific/DownsampleSinglePassMipChainPass.cpp

```diff
@@ -0,0 +1,297 @@
+            AZ_Assert(m_globalAtomicBuffer, "DownsampleSinglePassMipChainPass Building Global Atomic Buffer failed.")
+        }
+
+        void DownsampleSinglePassMipChainPass::InitializeIndices()
+        {
+            if (!m_shaderResourceGroup)
+            {
+                return;
+            }
+            const ShaderResourceGroup& srg = *m_shaderResourceGroup;
+
+            m_mipsIndex = srg.FindShaderInputConstantIndex(Name("m_mips"));
+            m_numWorkGroupsIndex = srg.FindShaderInputConstantIndex(Name("m_numWorkGroups"));
+            m_workGroupOffsetIndex = srg.FindShaderInputConstantIndex(Name("m_workGroupOffset"));
+            m_imageSizeIndex = srg.FindShaderInputConstantIndex(Name("m_imageSize"));
+            m_inputOutputImageIndex = srg.FindShaderInputImageIndex(Name("m_imageDestination"));
+            m_mip6ImageIndex = srg.FindShaderInputImageIndex(Mip6Name);
+            m_globalAtomicIndex = srg.FindShaderInputBufferIndex(GlobalAtomicName);
+            m_indicesAreInitialized = true;
+        }
+
+        void DownsampleSinglePassMipChainPass::GetInputInfo()
+        {
+            // Get the input/output mip chain attachment for this pass (at binding 0)
+            AZ_Assert(
+                GetInputOutputCount() > 0, "[DownsampleSinglePassMipChainPass '%s']: must have an input/output", GetPathName().GetCStr());
+            PassAttachment* attachment = GetInputOutputBinding(0).GetAttachment().get();
+
+            if (attachment)
+            {
+                m_mipLevelCount = attachment->m_descriptor.m_image.m_mipLevels;
+                m_inputImageSize[0] = attachment->m_descriptor.m_image.m_size.m_width;
+                m_inputImageSize[1] = attachment->m_descriptor.m_image.m_size.m_height;
+            }
+        }
+
+        void DownsampleSinglePassMipChainPass::CalculateBaseSpdImageSize()
+        {
+            const auto adjust = [&] (uint32_t value) -> uint32_t
+            {
+                const auto logValue = static_cast<uint32_t>(floorf(log2f(value * 1.f)));
+                if (static_cast<uint32_t>(1 << logValue) == value)
+                {
+                    return value;
+                }
+                else
+                {
+                    return static_cast<uint32_t>(1 << (logValue + 1));
+                }
+            };
+            m_baseSpdImageSize[0] = adjust(m_inputImageSize[0]);
+            m_baseSpdImageSize[1] = adjust(m_inputImageSize[1]);
+        }
+
+        void DownsampleSinglePassMipChainPass::BuildPassAttachment()
+        {
+            // Build "Mip6" image attachment.
+            {
+                const auto width = GetMax<uint32_t>(1, m_baseSpdImageSize[0] >> GloballyCoherentMipIndex);
+                const auto height = GetMax<uint32_t>(1, m_baseSpdImageSize[1] >> GloballyCoherentMipIndex);
+                m_mip6ImageDescriptor =
```

Comment:
Should the format chosen for this descriptor be based on the input image format?

Reply:
SPD stores each mip level values into groupshared float arrays except for mip level 6, and do into this "Mip6" image for mip level 6.  In the view of coherency, I think the precision of the stored value should be the same 32 bit float.

Reply:
I added a comment in the code to explain this.